src/TriggerActivityMakerTrackFinding.cpp

```diff
@@ -0,0 +1,303 @@
+	bool new_tp = 1;
+	for (auto tp_sel : win_adj_max.inputs) {
+	  if (tp.channel == tp_sel.channel) new_tp = 0;
+	}
+	if (new_tp) m_current_window.add(tp);
+      }
+      
+      // check adjacency -> win_adj_max now contains only those tps that make the track
+      win_adj_max = check_adjacency();
+      if (win_adj_max.inputs.size() > 0) {
+    	
+	adj_pass = 1;
+	ta_found = 1;
+	ta_count++;
+	if (ta_count % m_prescale == 0){
+	  
+	  output_ta.push_back(construct_ta(win_adj_max));
+	}
+      }
+      else ta_found = 0;
+    }
+    if (adj_pass) m_current_window.reset(input_tp);
+  }
+  
+  // if adjacency logic is not true, slide the window along using the current TP.
+  if (!window_filling && !adj_pass) {
+    m_current_window.move(input_tp, m_window_length);  
+  }
+  
+  using namespace std::chrono;
+  // If this is the first TP of the run, calculate the initial offset:
+  if (m_primitive_count == 0){
+    m_initial_offset = duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count() - input_tp.time_start*16*1e-6;
+  }
+  
+  // Update OpMon Variable(s)
+  uint64_t system_time = duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count();
+  uint64_t data_time = input_tp.time_start*16*1e-6;                              // Convert 62.5 MHz ticks to ms
+  m_data_vs_system_time.store(fabs(system_time - data_time - m_initial_offset)); // Store the difference for OpMon*/
+  m_primitive_count++;
+  
+  return;
+}
+
+void
+TriggerActivityMakerTrackFinding::configure(const nlohmann::json& config)
+{
+  if (config.is_object()) {
+    if (config.contains("window_length"))
+      m_window_length = config["window_length"];
+    if (config.contains("adj_tolerance"))
+      m_adj_tolerance = config["adj_tolerance"];
+    if (config.contains("adjacency_threshold"))
+      m_adjacency_threshold = config["adjacency_threshold"];
+    if (config.contains("print_tp_info"))
+      m_print_tp_info = config["print_tp_info"];
+    if (config.contains("prescale"))
+      m_prescale = config["prescale"]; 
+  }
+}
+
+TriggerActivity
+TriggerActivityMakerTrackFinding::construct_ta(TPWindow win_adj_max) const
+{
+  
+  TriggerActivity ta;
+  
+  TriggerPrimitive last_tp = win_adj_max.inputs.back();
+  
+  ta.time_start = last_tp.time_start;
+  ta.time_end = last_tp.time_start+last_tp.time_over_threshold;
+  ta.time_peak = last_tp.time_peak;
+  ta.time_activity = last_tp.time_peak;
+  ta.channel_start = last_tp.channel;
+  ta.channel_end = last_tp.channel;
+  ta.channel_peak = last_tp.channel;
+  ta.adc_integral = win_adj_max.adc_integral;
+  ta.adc_peak = last_tp.adc_integral;
+  ta.detid = last_tp.detid;
+  ta.type = TriggerActivity::Type::kTPC;
+  ta.algorithm = TriggerActivity::Algorithm::kTrackFinding;
+  ta.inputs = win_adj_max.inputs;
+
+  for( const auto& tp : ta.inputs ) {
+    ta.time_start = std::min(ta.time_start, tp.time_start);
+    ta.time_end = std::max(ta.time_end, tp.time_start+tp.time_over_threshold);
```

Comment:
Ditto r.e. moving away from this definition of `ta.time_end`. 
```
std::max(ta.time_end, tp.time_start);
```
is probably what we want here

Reply:
corrected!